budget-service/src/main/java/org/java/mentorship/budget/persistence/BudgetRepository.java

```diff
@@ -0,0 +1,66 @@
+package org.java.mentorship.budget.persistence;
+
+import lombok.RequiredArgsConstructor;
+import org.java.mentorship.budget.domain.BudgetEntity;
+import org.springframework.jdbc.core.JdbcTemplate;
+import org.springframework.jdbc.core.RowMapper;
+import org.springframework.stereotype.Repository;
+
+import java.util.List;
+
+@Repository
+@RequiredArgsConstructor
+public class BudgetRepository {
+
+    private final JdbcTemplate jdbcTemplate;
+    private final RowMapper<BudgetEntity> rowMapper;
+
+    public BudgetEntity save(final BudgetEntity budgetEntity) {
+        jdbcTemplate.update(
+                "INSERT INTO budgets (user_id, name, maximum_allowed, interval, current_usage, transaction_id, account_id) VALUES(?,?,?,?,?,?,?)",
+                budgetEntity.getUserId(),
+                budgetEntity.getName(),
+                budgetEntity.getMaximumAllowed(),
+                budgetEntity.getInterval().name(),
+                budgetEntity.getCurrentUsage(),
```

Comment:
Saving the current usage is tricky.
This means that every time we insert a transaction, this currentUsage needs to be updated.
And it also needs to be reset when the interval passes.
This would be a major headache.

Instead, I believe this field should be a "virtual" field. It doesn't have a database column, instead it is computed on every request by summing up all the user's transactions by a category in a given period of time.